### PR TITLE
IceBox Atmospherics Expansion - Infiltrate and Exfiltrate

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Hey there,

image

image

Ok Mr. Diamond-And-Win. Watch and learn.

I've seen like three screenshots of people working on revamping IceBox's atmospherics but literally none of them have followed through on it, so I've decided to make Multi-Z Atmospherics (boo). Also, above screenshots.

HOWEVER, it's not what you think. I play an atmospheric technician in game, and I tried to get a consistent strike between having the bare basics set up- while still having plenty of freedom to set up stuff with as you wish and get all the stuff from the atmospherics loop. Here's the pretties:

New Loop:

image

New Hallway-Storage Sector:

image

The Project Zones and the "Minor Loop":

image

There's a little small portion hidden on the third z-level, but you can see a little bit peeking out. I have also tweaked the placement of the AI Satellite + transit tube, some things in the surrounding maintenance/chasms, and the incinerator room a bit.

This does add some complexity to the atmospherics loop- but I think that the simplicity it offers later on when you interact with the minor loop on the lower z-level can not be understated. I think toying around with the "Box" atmospherics system and loop is just... too much to handle right now, so I thought to just build off the current loop and make the "minor" loop.

I also played around with catwalks and decals- and I think the decals add a little more context to what the pipes do. I played around some with the industrial theme (having catwalks over the HFR area and the "minor loop"), and I like the way it comes off in game. I definitely took inspiration from stuff like _Chernobyl_ as far as having catwalks directly above the optimal placement for the HFR. I decided to rely on shutters for more open flow rather than have obtuse windows.

Why "infiltrate" and "exfiltrate" for the minor loops? It sounds like "filtrate", which is something that you do in atmos (harvest and filter the gases you get from your various machinery). I also _really_ wanted to ensure that any new terminology I introduce would not be used in any other application in atmospherics (stuff like "vent", "input/output", etc will simply not do), so I do not think I will change it unless a good idea is offered up by you fine people.

## Why It's Good For The Game

Simply put: this is the atmospherics I want to see, and I think it's a good change. It adds features that IceBox simply did not have at all (making it worse than other maps), and pencils out the boundaries of atmospherics mapping+planning. 

## Changelog

:cl:
add: Nanotrasen has now installed an HFR Room and an Atmospherics Project Room on IceBoxStation. The Atmospherics Storage Room has also had catwalks installed in to accomodate these rooms uninstalled around it.
add: The Atmospherics Loop on IceBoxStation has undergone some minor modifications to accomodate "feeding" these new rooms. Notably, there is now a "minor" loop on the lower Z-Level that you can push and pull gases towards to your (probable) heart's content.
add: The AI Satellite, Incinerator, and some sections of Maintenance have been shuffled around on IceBoxStation to accomodate these changes.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not having the hfr was intentional, because it got too hot and icebox's planetary atmos couldn't contain it
That ought? to be solved now, but you should check to see what a delam is like

**Comment by san7890** [comment]
> Not having the hfr was intentional, because it got too hot and icebox's planetary atmos couldn't contain it That ought? to be solved now, but you should check to see what a delam is like

I have no idea how to test the HFR but I'll look up a guide. If it doesn't work at all I will remove the HFR core materials and leave the room empty to spur on the coders to fix it.

**Comment by Barnet2** [comment]
For the main atmospheric room, could you have layer manifolds where the gas pumps for the main gasses are for the pure line? It's done on Delta and Meta, and it is nice to not have to replace them when you want plasma for two different things.

**Comment by san7890** [comment]
> For the main atmospheric room, could you have layer manifolds where the gas pumps for the main gasses are for the pure line? It's done on Delta and Meta, and it is nice to not have to replace them when you want plasma for two different things.

Yeah, I get that sentiment since I did the same thing myself when I was really into atmospherics. I'll probably pass that in as well.

**Comment by san7890** [comment]
After a bit of struggling with the wiki, I have ascertained that I do not have any clue how to set up the HFR (is it disabled on Ice Box/Planetary Maps?). Please feel free to pull this branch onto your local machine and see if you can test it out if you're so willing, or let me know on Discord `@san7890#7890` if you wanna walk me through it- I will not understand it any other way.

**Comment by jwc1015** [comment]
> Not having the hfr was intentional, because it got too hot and icebox's planetary atmos couldn't contain it
> That ought? to be solved now, but you should check to see what a delam is like

Don’t we have the buffed thermomachines now without ballast? Shouldn’t be all that hard to deal with provided an atmos tech uses them afaik.

**Comment by Colovorat** [comment]
> Not having the hfr was intentional, because it got too hot and icebox's planetary atmos couldn't contain it 

ARE YOU SERIOUS? HAHAHAHAHA

**Comment by D4C-420** [comment]
https://www.youtube.com/watch?v=-vDnks-YIq0
Here ya go. HFR delam on icebox. 
Some notes
1) first time doing this kind of thing so uhh sorry for the poor quality
2)I could not figure out how to get this PR's version of icebox onto my local machine and I decided to just record this on icebox as it currently is.
3) this is filmed with the highest tier possible gasses for fuel and moderator. Your more likely to see lower quality fuels and such

TLDR. Icebox is perfectly habitable after the HFR blows up. The damage done is about on par with an sm explosion of decent size.

**Comment by san7890** [comment]
> https://www.youtube.com/watch?v=-vDnks-YIq0 Here ya go. HFR delam on icebox. Some notes
> 
> 1. first time doing this kind of thing so uhh sorry for the poor quality
>    2)I could not figure out how to get this PR's version of icebox onto my local machine and I decided to just record this on icebox as it currently is.
> 2. this is filmed with the highest tier possible gasses for fuel and moderator. Your more likely to see lower quality fuels and such
> 
> TLDR. Icebox is perfectly habitable after the HFR blows up. The damage done is about on par with an sm explosion of decent size.

Thank you so much!!!

**Comment by san7890** [comment]
> Edit: I looked into it and you'll need to MAJORLY overhaul the boxstation atmos loop to be bigger to accomodate for that change. I think that's too out-of-scope for this PR (read: I don't wanna waste the time doing it), but I'll keep it in mind for the future. Thank you for the suggestion though.l

Disregard this, the latest commit has now mapped this change, with great thanks to the atmospherics technician's community for the advice. The PR Body will be updated with a screenshot of the latest revision (it required I remove a port and a radsuit, but you guys have plenty of those).

**Comment by notfrying1pans** [comment]
👀 

image

**Comment by notfrying1pans** [comment]
btw there are unmentioned science changes in this pr

image

**Comment by san7890** [comment]
> btw there are unmentioned science changes in this pr
> 
> image

oh fuck i accidentally reverted #67778 one second

**Review by Maurukas — CHANGES_REQUESTED** [review]
You have solved the problem of Ice's terrible atmos projects room. 

Power looks good. Station atmos loops look good. No ATs or obvious map issues. The pipes work. The layout looks usable, spacious and well-decalled to me.

Now onto the stuff I found + suggestions:

<details> 
<summary>Suggestions</summary>

Add a maint access door to the right side room (instead of pipe under R-wall)
image

Consider a window here, or some kind of visual indication that this scrubber vents directly outside and not to the waste loop:
image
image


Not a suggestion, just want to note that I like this area and it is a great use of multi-z

image


</details>

<details>
<summary> Missing shutter tag here</summary>
Would also benefit from an additional light:

image

</details>

<details>
<summary>Disposal chute embedded in an R-wall</summary>

image
image

</details>

<details>
<summary> Big Uh-Oh - Layer adapter is bad here</summary>

This is crosslinking the airmix (cyan) with the pink waste pipe that connects HFR/Crystalizer/incinerator to the internal atmos waste loop. This will vent (slowly) the N2 tank if the pump in the incinerator is turned on. If waste gas is pumped into this pipe it will reach the airmix tank and contaminate the station's air supply.

image


</details>
